### PR TITLE
Nav to plant detail 🌱 

### PR DESCRIPTION
**Pull request by mnbpdx**
Set up basic nav from HomeScreen() to DetailScreen().

We probably should be using [compose nav](https://developer.android.com/develop/ui/compose/navigation), but for now we've kept it simple and are just `when`ing on a `NavigationViewModel.destination : State<Destination>`.

### CODE REVIEW

File: app/src/main/java/com/mnbpdx/gardenbook/ui/AppBars.kt [hunk 1]
```diff
@@ -54,10 +57,29 @@ internal fun GardenBookBottomAppBar(
 
 @ExperimentalMaterial3Api
 @Composable
-internal fun GardenBookTopAppBar() {
+internal fun GardenBookTopAppBar(
+    destination: Destination,
+    onUpButtonPress: () -> Unit,
```
Comment (tmprevo) [inline app/src/main/java/com/mnbpdx/gardenbook/ui/AppBars.kt]:
```suggestion
    onArrowBackPress: () -> Unit,
```

File: app/src/main/java/com/mnbpdx/gardenbook/ui/home/HomeScreen.kt [hunk 2]
```diff
@@ -20,62 +21,69 @@ import androidx.compose.material3.Text
...
 internal fun HomeScreenContent(
-    onPlantCardPress: () -> Unit
+    onPlantCardPress: (String) -> Unit,
 ) {
     Scaffold(
-        topBar = { GardenBookTopAppBar() },
+        topBar = {
+            GardenBookTopAppBar(
+                destination = Destination.HomeScreen,
+                onUpButtonPress = { }
```
Comment (tmprevo) [inline app/src/main/java/com/mnbpdx/gardenbook/ui/home/HomeScreen.kt]:
```suggestion
                onArrowBackPress = { }
```

File: app/src/main/java/com/mnbpdx/gardenbook/ui/detail/DetailScreen.kt [hunk 3]
```diff
@@ -1,44 +1,86 @@
...
+    )
+}
 
+@ExperimentalMaterial3Api
 @Composable
-internal fun DetailScreen(plant: String) {
-        Surface(modifier = Modifier.fillMaxSize()) {
+internal fun DetailScreenContent(
+    plantName: String,
+    onUpButtonPress: () -> Unit,
```
Comment (tmprevo) [inline app/src/main/java/com/mnbpdx/gardenbook/ui/detail/DetailScreen.kt]:
```suggestion
    onArrowBackPress: () -> Unit,
```

File: app/src/main/java/com/mnbpdx/gardenbook/ui/detail/DetailScreen.kt [hunk 4]
```diff
@@ -1,44 +1,86 @@
...
+) {
+    Scaffold(
+        topBar = {
+            GardenBookTopAppBar(
+                destination = Destination.DetailScreen(plantName), onUpButtonPress = onUpButtonPress
```
Comment (tmprevo) [inline app/src/main/java/com/mnbpdx/gardenbook/ui/detail/DetailScreen.kt]:
```suggestion
                destination = Destination.DetailScreen(plantName), onArrowBackPress = onArrowBackPress
```

File: app/src/main/java/com/mnbpdx/gardenbook/ui/detail/DetailScreen.kt [hunk 5]
```diff
@@ -1,44 +1,86 @@
...
 
+@OptIn(ExperimentalMaterial3Api::class)
 @Preview
 @Composable
 private fun DetailScreenPreview() {
     GardenBookTheme {
-        DetailScreen(plant = "Plant Name")
+        DetailScreenContent(
+            plantName = "Plant Name",
+            onUpButtonPress = {},
```
Comment (tmprevo) [inline app/src/main/java/com/mnbpdx/gardenbook/ui/detail/DetailScreen.kt]:
```suggestion
            onArrowBackPress = {},
```

File: app/src/main/java/com/mnbpdx/gardenbook/ui/home/HomeScreen.kt (hunk 2 above)
Comment (tmprevo) [inline app/src/main/java/com/mnbpdx/gardenbook/ui/home/HomeScreen.kt]:
```suggestion
                onArrowBackPress = { }
```

File: app/src/main/java/com/mnbpdx/gardenbook/ui/AppBars.kt [hunk 6]
```diff
@@ -54,10 +57,29 @@ internal fun GardenBookBottomAppBar(
...
     TopAppBar(
         title = {
             Text(text = stringResource(R.string.app_name))
         },
+        navigationIcon = {
+            when (destination) {
+                Destination.HomeScreen -> {}
+
+                is Destination.DetailScreen -> {
+                    IconButton(onClick = onUpButtonPress) {
```
Comment (tmprevo) [inline app/src/main/java/com/mnbpdx/gardenbook/ui/AppBars.kt]:
```suggestion
                    IconButton(onClick = onArrowBackPress) {
```

File: app/src/main/java/com/mnbpdx/gardenbook/ui/detail/DetailScreen.kt [hunk 7]
```diff
@@ -1,44 +1,86 @@
...
+internal fun DetailScreen(
+    plantName: String,
+    onBackPress: () -> Unit,
+) {
+    BackHandler {
+        onBackPress()
+    }
+
+    DetailScreenContent(
+        plantName = plantName, onUpButtonPress = onBackPress
```
Comment (tmprevo) [inline app/src/main/java/com/mnbpdx/gardenbook/ui/detail/DetailScreen.kt]:
```suggestion
        plantName = plantName, onArrowBackPress = onBackPress
```

File: app/src/main/java/com/mnbpdx/gardenbook/ui/theme/Type.kt [hunk 8]
```diff
@@ -15,7 +15,7 @@ val provider = GoogleFont.Provider(
 
 val bodyFontFamily = FontFamily(
     Font(
-        googleFont = GoogleFont("Cutive"),
+        googleFont = GoogleFont(name = "Cutive"),
```
Comment (tmprevo) [inline app/src/main/java/com/mnbpdx/gardenbook/ui/theme/Type.kt]: Remove parameter 'name' or add to line 25

**Review by Tearemuptara — APPROVED** [review]
Looks amazing! Added some comments ♥

File: app/src/main/java/com/mnbpdx/gardenbook/ui/theme/Type.kt (hunk 8 above)
Comment (mnbpdx) [inline app/src/main/java/com/mnbpdx/gardenbook/ui/theme/Type.kt]:
```suggestion
        googleFont = GoogleFont("Cutive"),
```

File: app/src/main/java/com/mnbpdx/gardenbook/ui/home/HomeScreen.kt (hunk 2 above)
Comment (mnbpdx) [inline app/src/main/java/com/mnbpdx/gardenbook/ui/home/HomeScreen.kt]:
```suggestion
                onArrowBackPress = { }
```

**Pull request merged**